tiledb/type/range/range.h

```diff
@@ -257,6 +259,102 @@ class Range {
   uint64_t partition_depth_;
 };
 
+/**
+ * Performs correctness checks for a valid range. If any validity checks fail
+ * an error status is returned.
+ *
+ * @param range The range to check.
+ * @return Status of the checks.
+ */
+template <
+    typename T,
+    typename std::enable_if<std::is_arithmetic<T>::value>::type* = nullptr>
+Status check_range_is_valid(const Range& range) {
+  // Check has data.
+  if (range.empty())
+    return Status_Error("Range is empty");
```

Comment:
Should the status error is specific to the `Range` class (same everywhere in the code)?

Reply:
Added `Status_RangeError`